sklearn/neighbors/binary_tree.pxi

```diff
@@ -1138,7 +1138,9 @@ cdef class BinaryTree:
                 int(self.n_splits),
                 int(self.n_calls),
                 self.dist_metric,
-                self.sample_weight_arr)
+                self.sample_weight_arr,
+                self.sample_weight,
+                self.sum_weight)
```

Comment:
I think you can calculate this when unpickling without storing it.

Reply:
@jnothman When only storing just self.sample_weight_arr, it fails too.

Reply:
@jnothman Any ideas as to why it is still failing?